### DISCUSSION TITLE
Coding Standards

### DISCUSSION THREAD

**Discussion by Zaba505** [z5labs/evrys#60] [Ideas]
This discussion is to serve as a home for collecting and recording some initial coding standards that contributions to evrys are expected to follow.

### Standards
- All public methods/functions must wrap any errors they encounter before returning
- All public values/types/methods/functions should be well documented
- Never create a util, common, or other generic name package
- Don't include method/function calls in assert statements for better readability

**Comment by Zaba505** [comment]
@erictg What are your thoughts on `panic`? Should we simply just outlaw its usage unless necessary?

**Reply by erictg** [reply]
I don't think there's ever a good reason to panic. The error should always be returned up the stack and handled. If the intention is for the program to exit on a fatal error, it is better to exit with a code to than to panic. For example, if we want the program to crash from a developer config error, it is still better to return errors up to the cli layer then exit with a code than to panic out down stack.
